test/src/unittests/tonal/test_pitchsaliencefunction.py

```diff
@@ -0,0 +1,223 @@
+        guessUnvoiced = True
+
+        run_windowing = Windowing(type='hann', zeroPadding=3*frameSize) # Hann window with x4 zero padding
+        run_spectrum = Spectrum(size=frameSize * 4)
+        run_spectral_peaks = SpectralPeaks(minFrequency=1,
+                                           maxFrequency=20000,
+                                           maxPeaks=100,
+                                           sampleRate=sampleRate,
+                                           magnitudeThreshold=0,
+                                           orderBy="magnitude")
+        run_pitch_salience_function = PitchSalienceFunction()
+        run_pitch_salience_function_peaks = PitchSalienceFunctionPeaks()
+        run_pitch_contours = PitchContours(hopSize=hopSize)
+        run_pitch_contours_melody = PitchContoursMelody(guessUnvoiced=guessUnvoiced,
+                                                        hopSize=hopSize)
+
+        signalSize = frameSize*10
+        # Here are generate sine waves for each note of the scale, e.g. C3 is 130.81 Hz, etc
+        c3 = 0.5 * numpy.sin((array(range(signalSize))/44100.) * 130.81 * 2*math.pi)
+        d3 = 0.5 * numpy.sin((array(range(signalSize))/44100.) * 146.83 * 2*math.pi)
+        e3 = 0.5 * numpy.sin((array(range(signalSize))/44100.) * 164.81 * 2*math.pi)
+        f3 = 0.5 * numpy.sin((array(range(signalSize))/44100.) * 174.61 * 2*math.pi)
+        g3 = 0.5 * numpy.sin((array(range(signalSize))/44100.) * 196.00 * 2*math.pi)                                
+        a3 = 0.5 * numpy.sin((array(range(signalSize))/44100.) * 220.00 * 2*math.pi)
+        b3 = 0.5 * numpy.sin((array(range(signalSize))/44100.) * 246.94 * 2*math.pi)
+        c4 = 0.5 * numpy.sin((array(range(signalSize))/44100.) * 261.63 * 2*math.pi)
+    
+        # This signal is a "major scale ladder"
+        scale = concatenate([c3, d3, e3, f3, g3, a3, b3, c4])
+
+        # Now we are ready to start processing.
+        # 1. Load audio and pass it through the equal-loudness filter
+        audio = EqualLoudness()(scale)
+
+        # 2. Cut audio into frames and compute for each frame:
+        #    spectrum -> spectral peaks -> pitch salience function -> pitch salience function peaks
+        for frame in FrameGenerator(audio, frameSize=frameSize, hopSize=hopSize):
+            frame = run_windowing(frame)
+            spectrum = run_spectrum(frame)
+            peak_frequencies, peak_magnitudes = run_spectral_peaks(spectrum)
+            salience = run_pitch_salience_function(peak_frequencies, peak_magnitudes)
+            salience_peaks_bins, salience_peaks_saliences = run_pitch_salience_function_peaks(salience)
+                
+        expectedBins = [270., 150. ,80. ,30.]
+        expectedPeaks =  [0.09777679, 0.07822143, 0.06257715, 0.05006172]
+        self.assertAlmostEqualVectorFixedPrecision(expectedBins, salience_peaks_bins, 6)
+        self.assertAlmostEqualVectorFixedPrecision(expectedPeaks, salience_peaks_saliences, 6)
```

Comment:
Same here. We are testing PitchSalienceFunction, we can do it directly on its output. The tests for peaks are just an addition on top.

Perhaps we could have a npy with the expected matrix output if it isn't too large in filesize?